tests/psbt.rs

```diff
@@ -0,0 +1,702 @@
+fn parse_and_verify_keys(ext_priv: &ExtendedPrivKey, sk_path: &[(&str, &str)]) -> Vec<PrivateKey> {
+    let secp = SECP256K1;
+
+    let mut keys = vec![];
+    for (secret_key, derivation_path) in sk_path.iter() {
+        let wifi_priv = PrivateKey::from_wif(secret_key).expect("failed to parse key");
+
+        let path = derivation_path.into_derivation_path().expect("failed to convert derivation path");
+        let derived_priv = ext_priv.derive_priv(secp, &path).expect("failed to derive ext priv key").to_priv();
+        assert_eq!(wifi_priv, derived_priv);
+        keys.push(wifi_priv)
+    }
+    keys
+}
+
+/// Does the first signing according to the BIP, returns the signed PSBT. Verifies against BIP 174 test vector.
+fn signer_one_sign(psbt: Psbt, keys: &[PrivateKey]) -> Psbt {
+    let expected_psbt_hex = include_str!("data/sign_1_psbt_hex");
+    let expected_psbt = hex_psbt!(expected_psbt_hex).unwrap();
+
+    let psbt = sign(psbt, keys);
+
+    assert_eq!(psbt, expected_psbt);
+    psbt
+}
+
+/// Does the second signing according to the BIP, returns the signed PSBT. Verifies against BIP 174 test vector.
+fn signer_two_sign(psbt: Psbt, keys: &[PrivateKey]) -> Psbt {
+    let expected_psbt_hex = include_str!("data/sign_2_psbt_hex");
+    let expected_psbt = hex_psbt!(expected_psbt_hex).unwrap();
+
+    let psbt = sign(psbt, keys);
+
+    assert_eq!(psbt, expected_psbt);
+    psbt
+}
+
+/// Does the combine according to the BIP, returns the combined PSBT. Verifies against BIP 174 test vector.
+fn combine(mut this: Psbt, that: Psbt) -> Psbt {
+    let expected_psbt_hex = include_str!("data/combine_psbt_hex");
+    let expected_psbt = hex_psbt!(expected_psbt_hex).unwrap();
+
+    this.combine(that).expect("failed to combine PSBTs");
+
+    assert_eq!(this, expected_psbt);
+    this
+}
+
+/// Does the finalize step according to the BIP, returns the combined PSBT. Verifies against BIP 174
+/// test vector.
+fn finalize(psbt: Psbt) -> Psbt {
+    let expected_psbt_hex = include_str!("data/finalize_psbt_hex");
+    let expected_psbt = hex_psbt!(expected_psbt_hex).unwrap();
+
+    let psbt = finalize_psbt(psbt);
+
+    assert_eq!(psbt, expected_psbt);
+    psbt
+}
+
+/// Does the transaction extractor step according to the BIP, returns the combined PSBT. Verifies
+/// against BIP 174 test vector.
+fn extract_transaction(psbt: Psbt) -> Transaction {
+    let expected_tx_hex = include_str!("data/extract_tx_hex");
+
+    let tx = psbt.extract_tx();
+
+    let got = serialize_hex(&tx);
+    assert_eq!(got, expected_tx_hex);
+
+    tx
+}
+
+/// Combines two PSBTs lexicographically according to the BIP. Verifies against BIP 174 test vector.
+fn combine_lexicographically() {
+    let psbt_1_hex = include_str!("data/lex_psbt_1_hex");
+    let psbt_2_hex = include_str!("data/lex_psbt_2_hex");
+
+    let expected_psbt_hex = include_str!("data/lex_combine_psbt_hex");
+    let expected_psbt = hex_psbt!(expected_psbt_hex).unwrap();
+    
```

Comment:
This has trailing whitespace, the githook is not catching it, I can't work out why.